pkg/kms/localkms/localkms_writer.go

```diff
@@ -0,0 +1,69 @@
+/*
+ Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+ SPDX-License-Identifier: Apache-2.0
+*/
+
+package localkms
+
+import (
+	"encoding/base64"
+	"fmt"
+	"io"
+
+	"github.com/google/tink/go/subtle/random"
+
+	"github.com/hyperledger/aries-framework-go/pkg/storage"
+)
+
+// newWriter creates a new instance of local storage key storeWriter in the given store and for masterKeyURI
+func newWriter(kmsStore storage.Store, masterKeyURI string) io.Writer {
+	return &storeWriter{
+		storage:      kmsStore,
+		masterKeyURI: masterKeyURI,
+	}
+}
+
+// storeWriter struct to store a keyset in a local store
+type storeWriter struct {
+	storage      storage.Store
+	masterKeyURI string
+	// KeysetID is set when Write() is called
+	KeysetID string
+}
+
+// Write a marshaled keyset p in localstore with masterKeyURI prefix + randomly generated KeysetID
+func (l *storeWriter) Write(p []byte) (int, error) {
+	if l.masterKeyURI == "" {
+		return 0, fmt.Errorf("master key is not set")
+	}
+
+	const keySetIDLength = 32
+
+	baseID := l.masterKeyURI + "/"
+	ksID := ""
+
+	for {
+		// generate random ID
+		ksID = baseID + base64.URLEncoding.EncodeToString(random.GetRandomBytes(keySetIDLength))
```

Comment:
Why not just use crypto/rand? crypto/rand guarantees a "cryptographically secure random number generator" so there should be no collisions

Reply:
random.GetRandomBytes() is a utility function that calls crypto/rand under the hood, it saves the boilerplate code of creating and copying a byte array.